src/main/java/reposense/commits/CommitInfoExtractor.java

```diff
@@ -43,16 +43,21 @@
      */
     private static ArrayList<CommitInfo> parseGitLogResults(String gitLogResult) {
         ArrayList<CommitInfo> commitInfos = new ArrayList<>();
-        String[] rawLines = gitLogResult.split("\n");
+        String[] rawCommitInfos = gitLogResult.split(GitLog.COMMIT_INFO_DELIMITER);
 
-        if (rawLines.length < 2) {
+        if (rawCommitInfos.length < 2) {
             //no log (maybe because no contribution for that file type)
             return commitInfos;
         }
 
-        for (int i = 0; i < rawLines.length; i++) {
-            commitInfos.add(new CommitInfo(rawLines[i], rawLines[++i]));
-            i++; //to skip the empty line
+        // Starts from 1 as index 0 is always empty.
+        for (int i = 1; i < rawCommitInfos.length; i++) {
+            String rawCommitInfo = rawCommitInfos[i].replaceAll("\n+$", "");
```

Comment:
Should we use compile pattern instead of `String.replaceAll`?

Reply:
Wasn't aware of the performance penalty of continuously using String.replaceall. Changed to use compile patterns instead.